src/main/java/org/candlepin/subscriptions/marketplace/MarketplacePayloadMapper.java

```diff
@@ -196,9 +196,11 @@ protected boolean isSnapshotPAYGEligible(TallySnapshot snapshot) {
             .forEach(measurement -> {
                 UsageMeasurement usageMeasurement = new UsageMeasurement();
                 usageMeasurement.setValue(measurement.getValue());
-                usageMeasurement.setMetricId(productProfile.getMetricId());
+                usageMeasurement.setMetricId(productProfile.metricByProductAndUom(productId,
+                    productProfile.fromTallyMeasurementUom(measurement.getUom())));
```

Comment:
I find this lookup here just a touch awkward. Can you please change the signature of metricByProductAndUom to accept `TallyMeasurement.Uom` (and just call `fromTallyMeasurementUom` internally)?

Reply:
Fair.  Done.